amplitude.py

```diff
@@ -86,7 +88,9 @@ def partition_available_events (events):
 def is_partitioned (partition):
     return partition[1] != ""
 
-def process (events, batches = {"identify": [], "event": []}, is_last_call = True):
+def process (pool, events, batches = None, is_last_call = True):
+    if batches is None:
+        batches = {"identify": [], "event": []}
```

Comment:
Mutable default arguments are a footgun in general in python, but especially so here - we have to make sure to create a new dict for `batches` on each iteration rather than mutating it in place, because we're passing it off to another thread.

Reply:
Yep. Fwiw, the approach I took in #15 was to pass the thread the stringified payload rather than the raw object. The same thing would work here too right? (just asking for my own understanding, not as a suggestion)

Reply:
Yep, that should also work.